hazelcast-sql-core/src/main/java/com/hazelcast/sql/impl/calcite/opt/physical/index/IndexResolver.java

```diff
@@ -116,6 +118,28 @@ private IndexResolver() {
             return Collections.emptyList();
         }
 
+        List<RelNode> rels = new ArrayList<>(supportedIndexes.size());
+
+        if (filter == null) {
+            // There is no filter, still generate index scans for
+            // possible ORDER BY clause on the upper level
+            for (MapTableIndex index : supportedIndexes) {
```

Comment:
Even if there is a filter, we still should generate full scans for the non-involved indexes. Otherwise, some plans will not work. Do we have a test, where:
1. There is an index on column `a`
2. Condition on the same column
3. `ORDER BY` on the other column?

E.g. `SELECT a, b FROM table(index(a)) WHERE a=1 ORDER BY b`

Reply:
Fixed in https://github.com/hazelcast/hazelcast/pull/18013/commits/30dc043f41e9d68727328ffc0302413f07d7bb2e and added the test